### PR TITLE
Fixes #406 - introduce instance_type_id to ovirt_vm resource

### LINKED ISSUE #406: Implement `instance_type_id` on `ovirt_vm`

### PR DESCRIPTION
**Pull request by eslutsky**
## Please describe the change you are making

introduce instance_type_id to ovirt_vm

## Are you the owner of the code you are sending in, or do you have permission of the owner?

Yes

## The code will be published under the BSD 3 clause license. Have you read and understood this license?

Yes

### CODE REVIEW

File: internal/ovirt/resource_ovirt_vm.go [hunk 1]
```diff
@@ -521,6 +528,21 @@ func handleVMInitialization(
 	return diags
 }
 
+func handleVMInstanceTypeID(
+	_ ovirtclient.Client,
+	data *schema.ResourceData,
+	params ovirtclient.BuildableVMParameters,
+	diags diag.Diagnostics,
+) diag.Diagnostics {
+	if instance_type_id, ok := data.GetOk("instance_type_id"); ok {
```
Comment (janosdebugs) [inline internal/ovirt/resource_ovirt_vm.go:537]: Please rename to `instanceTypeID` to pass linting.

File: examples/resources/ovirt_vm_instance_type_id/provider.tf [hunk 2]
```diff
@@ -0,0 +1,20 @@
+terraform {
```
Comment (janosdebugs) [inline examples/resources/ovirt_vm_instance_type_id/provider.tf:1]: `ovirt_vm_instance_type_id` is not a separate resource and should not be added to the examples. This folder is used for doc generation.

File: internal/ovirt/resource_ovirt_vm_test.go [hunk 3]
```diff
@@ -371,6 +371,61 @@ resource "ovirt_vm" "foo" {
...
+	config := fmt.Sprintf(
+		`
+provider "ovirt" {
+	mock = true
+}
+
+resource "ovirt_vm" "foo" {
+	cluster_id  = "%s"
+	template_id = "%s"
+    name        = "test"
```
Comment (janosdebugs) [inline internal/ovirt/resource_ovirt_vm_test.go:397]: Tabs/spaces issues?

File: internal/ovirt/resource_ovirt_vm_test.go [hunk 4]
```diff
@@ -371,6 +371,61 @@ resource "ovirt_vm" "foo" {
...
+
+	resource.UnitTest(
+		t, resource.TestCase{
+			ProviderFactories: p.getProviderFactories(),
+			Steps: []resource.TestStep{
+				{
+					Config: config,
+					Check: resource.ComposeTestCheckFunc(
+						resource.TestCheckResourceAttr(
+							"ovirt_vm.foo",
```
Comment (janosdebugs) [inline internal/ovirt/resource_ovirt_vm_test.go:414]: If we can, let's check the underlying VM using `helper.GetClient().GetVM()`

**Review by janosdebugs — CHANGES_REQUESTED** [review]

File: examples/resources/ovirt_vm_instance_type_id/provider.tf (hunk 2 above)
Comment (eslutsky) [reply examples/resources/ovirt_vm_instance_type_id/provider.tf]: removed

**Review by janosdebugs — APPROVED** [review]

File: internal/ovirt/resource_ovirt_vm.go [hunk 5]
```diff
@@ -164,6 +164,12 @@ var vmSchema = map[string]*schema.Schema{
 		Optional:    true,
 		Description: "Enable or disable the serial console.",
 	},
+	"instance_type_id": {
```
Comment (engelmi) [inline internal/ovirt/resource_ovirt_vm.go:167]: Please update the docs via `go generate`

**Pull request merged**